File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func solution(str, bomb string) string {
	stack := []rune{}
	bombLast := rune(bomb[len(bomb)-1])
	bLen := len(bomb)

	for _, char := range str {
		stack = append(stack, char)
		if len(stack) >= bLen && stack[len(stack)-1] == bombLast {
			if string(stack[len(stack)-bLen:]) == bomb {
				stack = stack[:len(stack)-bLen]
			}
		}
	}

	if len(stack) == 0 {
		return "FRULA"
	}
	return string(stack)
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	str, _ := reader.ReadString('\n')
	bomb, _ := reader.ReadString('\n')

	str = strings.TrimSpace(str)
	bomb = strings.TrimSpace(bomb)

	fmt.Println(solution(str, bomb))
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
from sys import stdin

input = stdin.readline


def solution(string, bomb):

    stack = []
    bomb_last = bomb[-1]
    blen = len(bomb)
    for char in string:
        stack.append(char)
        if len(stack) >= len(bomb) and stack[-1] == bomb_last:
            if "".join(stack[-blen:]) == bomb:
                del stack[-blen:]
    return "".join(stack) if stack else "FRULA"


if __name__ == "__main__":
    string = input().rstrip()
    bomb = input().rstrip()

    print(solution(string, bomb))
